core/templates/dev/head/exploration_editor/EditorServices.js

```diff
@@ -1952,3 +2012,283 @@ oppia.factory('explorationWarningsService', [
     };
   }
 ]);
+
+oppia.factory('lostChangesService', ['utilsService', function(utilsService) {
+  var CMD_ADD_STATE = 'add_state';
+  var CMD_RENAME_STATE = 'rename_state';
+  var CMD_DELETE_STATE = 'delete_state';
+  var CMD_EDIT_STATE_PROPERTY = 'edit_state_property';
+
+  var getRulesList = function(value) {
```

Comment:
Please change 'value' to something more descriptive that explains what kinds of inputs this function takes.